src/main/java/com/flightstats/hub/channel/InternalChannelResource.java

```diff
@@ -114,18 +116,20 @@ public Response refresh(@QueryParam("all") @DefaultValue("true") boolean all) {
     @Path("{channel}")
     @DELETE
     public Response delete(@PathParam("channel") final String channelName) {
-        channelService.getChannelConfig(channelName, false)
+        //added sanitizing method for user input for channelname to prevent XSS Vulnerability
+        String sanitizedChannelName = Encode.forUriComponent(StringEscapeUtils.escapeHtml4(channelName));
+        channelService.getChannelConfig(sanitizedChannelName, false)
                 .orElseThrow(() -> {
-                    Response errorResponse = ChannelResource.notFound(channelName);
+                    Response errorResponse = ChannelResource.notFound(sanitizedChannelName);
                     throw new WebApplicationException(errorResponse);
                 });
 
         if (contentProperties.isChannelProtectionEnabled()) {
-            log.info("using internal localhost only to delete {}", channelName);
-            return LocalHostOnly.getResponse(uriInfo, () -> deleteChannel(channelName));
+            log.info("using internal localhost only to delete {}", sanitizedChannelName);
+            return LocalHostOnly.getResponse(uriInfo, () -> deleteChannel(sanitizedChannelName));
```

Comment:
Should we need sanitizedChannelName in all instances ? deleteChannel() method uses channelName , which should not be the encoded one I guess.

Reply:
Sanitized and normal should be the same for any valid channel names, I think. There's a validation in ChannelValidator that uses the following regex for name: `public static final String VALID_NAME = "^[a-zA-Z0-9_-]+$";` (in `ContentConstant.java`). So...I think using either one should be ok in this case.

Reply:
yes, Sanitized and Normal should work same for any valid Channel Names. This is only applied to Public method which are exposing APIs. 
This can be tried. That method is private and can be moved into HubUtils and update references accordingly.
The only worry I have here, is while putting Channel Name in Spoke/S3, it has be encoded in same way.

Reply:
I'm not sure I fully understand this -- I don't think we need to use the validation I mentioned anywhere new, since it's being used on creation and update of channel configs. I was just pointing out that I believe `A-Z` + `_` and `-` are all characters that would not be impacted by url encoding, so I think encoded would always equal unencoded at the point that we're interacting with S3 or spoke. I might be missing something?

Reply:
This Channel Validator is using above regex to validate the channel name against A-Z + _ and -. And Encode.forUriComponent - is only doing against ChannelName. So yes agree, it is safe to use. It wont break anything. Now I got whole point here mentioning about it in your first comment. 

@Haripriya99j  - we can go with this change, just remove StringEscapeUtils.escapeHtml4, which is not needed. Rest all  good.

Reply:
Yes Abhay, Even I also approached in this way only. Once it goes well, I will raise another PR request with above mentioned changes.

Reply:
Closing this PR

Reply:
Using "StringEscapeUtils.escapeHtml4" only fix can be resolved. I just tried with many combination of encode snippets but still this only resolving the issue.
And There's a validation in ChannelValidator that uses the following regex for name: public static final String VALID_NAME = "^[a-zA-Z0-9_-]+$"; (in ContentConstant.java).
Since the issue is in InternalChannelResource.java, we need to add separate validator there also so that only it can be fixed for vulnerabilities. And here suggesting to go with "StringEscapeUtils.escapeHtml4".